module/zfs/zio.c

```diff
@@ -897,6 +897,7 @@ zfs_blkptr_verify_log(spa_t *spa, const blkptr_t *bp,
 	va_list adx;
 	char buf[256];
 
+	dprintf_bp(bp, "blkptr at %p dprintf_bp():", bp);
```

Comment:
This could potentially be very chatty for seriously a damaged block pointer since we call `zfs_blkptr_verify_log` for every failed verification check in `zfs_blkptr_verify`.  It would be nice to only log this information once for each block pointer.  What if we instead logged the full bp at the end of `zfs_blkptr_verify()` when `errors > 0`.  And since this wouldn't be logged in the `BLK_VERIFY_HALT` case, we could also log it in the `case BLK_VERIFY_HALT:` block in `zfs_blkptr_verify_log()`.

Note that this additional logging would also require `ZFS_DEBUG_DPRINTF` be set.

Reply:
Fixed!